ament_cmake_core/cmake/core/all.cmake

```diff
@@ -53,12 +53,23 @@ _set_cmake_policy_to_new_if_available(CMP0020)
 _set_cmake_policy_to_new_if_available(CMP0021)
 _set_cmake_policy_to_new_if_available(CMP0022)
 _set_cmake_policy_to_new_if_available(CMP0023)
+_set_cmake_policy_to_new_if_available(CMP0042)
```

Comment:
> I am not sure that it is a good idea to set "newer" policies since it will result in different behavior depending on the CMake version used.

In general you may have a point. In this particular case I don't think it is an issue. I'm fine updating the required version, but I think we should only do it for a good reason because it will make it harder for people on Ubuntu LTS to try out our software for the next few years.

> On the other hand the patch sets MACOSX_RPATH explicitly to ON so we don't need this policy at all, right?

In my testing it was neither sufficient to just set the policy or just set the setting, both were required, otherwise it complained, which I think is a bit of a bug, but I haven't completely convinced myself of that yet. I suppose the other option is to set it to `OLD`, but I think this is a better alternative.